ui/src/components/InputDialog/RoleDialog.tsx

```diff
@@ -0,0 +1,62 @@
+import React from 'react';
+
+import { useLedger } from '@daml/react';
+
+import { InputDialog } from './InputDialog';
+
+import { Template } from '@daml/types';
+import { RoleRequestTemplates, RoleKind } from '../../context/RolesContext';
+
+interface RequestProps<T extends RoleRequestTemplates> {
+  fields: any;
```

Comment:
Would the following work?
 
```suggestion
interface RequestProps<T extends RoleRequestTemplates, K extends { [key: string]: any }> {
  fields: Record<keyof K, Field>;
```

This is how `fields` is typed in InputDialog. Don't spend too much time on this if it's not an easy win, but it's always nice to not rely on `any` when we can

Reply:
Hah, of course, even the suggestion relies on `any` a bit anyway :p

Reply:
lol yep. I tried the above, which I then had to add the same constraint to `RoleRequestDialog` below...which then gave me this error for `InputDialog`'s `defaultValue` field:
```
tsserver 2322] [E] Type '{ operator: string; }' is not assignable to type 'K'.
  '{ operator: string; }' is assignable to the constraint of type 'K', but 'K' could be instantiated with a different subtype of constraint '{ [key: string]: any; }'.
```

I based this off of `ServiceDialog.tsx`, this may be why it used `any`...happy to try something else though, I'm curious why it won't work like this.

Reply:
Hm yep, off the top of my head, not sure how to work around the subtype constraint. Then this is good for now